app/src/main/java/com/example/dinhh/soundscape/device/Player.kt

```diff
@@ -39,11 +45,24 @@ class PlayerImpl: Player {
         }.andThen(onPlayComplete())
     }
 
+    override fun loopSound(){
+        isLooping = true
+    }
+
+    override fun stopLoop(){
+        isLooping = false
+    }
+
     override fun onPlayComplete(): Completable {
 
         return Completable.create {complete ->
             mediaPlayer.setOnCompletionListener {
-                complete.onComplete()
+                if (!isLooping) {
+                    complete.onComplete()
+                }else{
+                    mediaPlayer.seekTo(0)
+                    mediaPlayer.start()
```

Comment:
Not sure it's the right way to do this. Try this

```
fun setPlayerLooping((isLooping: bool): Completable {
  return Completable.fromAction {
  mediaPlayer.setLooping(isLooping)
 }
}
```

Reply:
There doesn't seem to be setLooping function in the Kotlin version of MediaPlayer.